ttnn/cpp/ttnn/operations/experimental/cnn/convert_to_hwc/device/kernels/writer_convert_to_hwc.cpp

```diff
@@ -0,0 +1,98 @@
+// SPDX-FileCopyrightText: © 2025 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "dataflow_api.h"
+#include "debug/dprint_pages.h"
+
+constexpr uint32_t TILE_SIZE = 32;
+constexpr uint32_t ELEMENT_SIZE_BYTES = 2;
+constexpr uint32_t TILE_STICK_SIZE_BYTES = TILE_SIZE * ELEMENT_SIZE_BYTES;
+
+template <uint32_t StickSize, uint32_t PaddedStickSize, uint32_t NumSticks>
+FORCE_INLINE void copy_padded_sticks(uint64_t l1_read_addr, uint32_t& l1_write_addr) {
+    noc_async_read_one_packet_set_state(l1_read_addr, StickSize);
+    for (uint32_t row = 0; row < NumSticks; row++) {
+        noc_async_read_one_packet_with_state<true>(l1_read_addr, l1_write_addr);
```

Comment:
src and dst adr for this api are uint32_t as these work on a single tensix core you don't need noc adr part